micro_http/src/connection.rs

```diff
@@ -332,19 +333,25 @@ impl<T: Read + Write> HttpConnection<T> {
                         response_fully_written = true;
                     }
                 }
-                Err(io_error) => {
-                    return Err(ConnectionError::StreamError(io_error));
-                }
             }
         }
 
-        if response_fully_written {
+        if connection_closed {
+            self.clear_write_buffer();
+            return Err(ConnectionError::ConnectionClosed);
+        } else if response_fully_written {
             self.response_buffer.take();
         }
 
         Ok(())
     }
 
+    fn clear_write_buffer(&mut self) {
+        let response_queue_len = self.response_queue.len();
+        self.response_queue.drain(0..response_queue_len);
```

Comment:
This can just be `self.response_queue.clear()` or `self.response_queue.drain(..)`